locations/spiders/trader_joes_us.py

```diff
@@ -7,23 +8,34 @@ class TraderJoesUSSpider(Where2GetItSpider):
     item_attributes = {"brand": "Trader Joe's", "brand_wikidata": "Q688825"}
     api_brand_name = "traderjoes"
     api_key = "8559C922-54E3-11E7-8321-40B4F48ECC77"
+    api_filter = {"comingsoon": {"ne": "Yes"}}
+
+    def pre_process_data(self, location):
+        location.pop("location", None)  # Fix coordinate mis-detection
 
     def parse_item(self, item, location):
-        item["website"] = (
-            "https://locations.traderjoes.com/"
-            + item["state"].lower()
-            + "/"
-            + item["city"].lower().replace(" ", "-")
-            + "/"
-            + item["ref"]
-            + "/"
-        )
-        hours_string = ""
-        for day_abbrev, day in zip(["mon", "tues", "wed", "thurs", "fri", "sat", "sun"], DAYS):
-            if location.get(day_abbrev + "open") and location.get(day_abbrev + "close"):
-                hours_string = (
-                    f"{hours_string} {day}: " + location[day_abbrev + "open"] + "-" + location[day_abbrev + "close"]
-                )
-        item["opening_hours"] = OpeningHours()
-        item["opening_hours"].add_ranges_from_string(hours_string)
+        if location["warehouse"] == "1":
```

Comment:
There is Categories.INDUSTRIAL_WAREHOUSE which you could use here

Reply:
They're not TJ's branded warehouses